client-linuxapp/src/serviceinfo.rs

```diff
@@ -615,18 +687,30 @@ async fn process_serviceinfo_in(si_in: &ServiceInfo, si_out: &mut ServiceInfo) -
         }
     }
 
-    // Do SSH
+    // Perform SSH or password setup
     if active_modules.contains(&FedoraIotServiceInfoModule::SSHKey.into()) {
-        log::debug!("SSHkey module was active, installing SSH key");
-        if sshkey_user.is_none() || sshkey_key.is_none() {
-            bail!("SSHkey module missing username or key");
+        if sshkey_user.is_none() {
```

Comment:
To make this more readable and less complicated you can make a first `if` with the failure conditions, `bail` there, and then continue handling the rest of the possible options

``` 
if sshkey_user.is_none() || (sshkey.is_some() && sshkey_password.is_some()) {
   bail!();
}
// then we continue with the rest
if something else { 
}
```
but, I think that the story was about allowing to add password and/or sshkey for a user, so both things at the same time should also be allowed.

Reply:
You're right, I miswrote that. I didn't think about just having all the bails together, thank you for the idea